bulk_deploy/deploy_actions.rb

```diff
@@ -0,0 +1,408 @@
+require 'fileutils'
+require 'yaml'
+require_relative 'tsuru_api_client'
+require_relative 'tsuru_api_service'
+require_relative 'tsuru_deploy_client'
+
+class DeployActions
+
+  def initialize(options)
+    @options = options
+    @logger = Logger.new(STDOUT)
+    case options[:log_level].downcase
+      when 'debug'
+        @logger.level = Logger::DEBUG
+      when 'info'
+        @logger.level = Logger::INFO
+      when 'warn'
+        @logger.level = Logger::WARN
+      when 'error'
+        @logger.level = Logger::ERROR
+      when 'fatal'
+        @logger.level = Logger::FATAL
+      else
+        raise "Error: Unknown log level: #{options[:log_level]}"
+    end
+  end
+
+  def apply()
+    environment       = @options[:environment]
+    host_suffix       = @options[:host_suffix]
+    api_token         = @options[:api_token]
+    search_api_token  = @options[:search_api_token]
+    ssh_config        = @options[:ssh_config]
+    team_count        = @options[:team_count]
+    apps_per_team     = @options[:apps_per_team]
+    users_per_team    = @options[:users_per_team]
+    units_per_app     = @options[:units_per_app]
+    state_file        = @options[:state_file]
+
+    tsuru_home = '/tmp/tsuru_tmp'
```

Comment:
Using a fixed temporary path might deal to problems (wrong state from previous executions, concurrent execution...). Is it not better to create a system provided random new directory, [as we do in the tsuru-ansible spec](https://github.com/alphagov/tsuru-ansible/blob/master/spec/tsuru_helper.rb#L19)?